scripts/irods/test/test_ichmod.py

```diff
@@ -207,6 +207,43 @@ class Access(Enum):
                         if access_level.value in range(check[0].value, check[1].value + 1):
                             self.user.assert_icommand(params.get('cmd'), *check[2])
 
+    def test_ichmod_does_not_bypass_permission_model_for_data_object__issue_6579(self):
+        data_object = 'testFile'
+        user_permissions = ['null', 'read', 'write']
+
+        for permission in user_permissions:
+            with self.subTest(permission):
+                user_cur_dir, _, _ = self.user.run_icommand('ipwd')
+                self.user.assert_icommand(f'itouch {data_object}')
+                self.user.assert_icommand(f'ichmod {permission} {self.user.username} {data_object}')
+                out, err, ec = self.user.run_icommand(f'ichmod own {self.user.username} {data_object}')
+
+                self.assertEqual(ec, 8)
+                self.assertIn('CAT_NO_ACCESS_PERMISSION', err)
+                self.assertEqual(len(out), 0)
+
+                # Remove data object or error will be thrown on cleanup
+                self.admin.assert_icommand(f'ichmod -M own {self.user.username} {os.path.join(user_cur_dir.rstrip(), data_object)}')
```

Comment:
no need for `ipwd` or `user_cur_dir`...  we have `self.user.session_collection`...